src/main/java/io/druid/data/input/impl/StringInputRowParser.java

```diff
@@ -35,20 +36,35 @@
  */
 public class StringInputRowParser implements ByteBufferInputRowParser
 {
+  private final static Charset defaultCharset = Charsets.UTF_8;
+
   private final ParseSpec parseSpec;
   private final MapInputRowParser mapParser;
   private final Parser<String, Object> parser;
+  private final Charset charset;
 
   private CharBuffer chars = null;
 
   @JsonCreator
   public StringInputRowParser(
-      @JsonProperty("parseSpec") ParseSpec parseSpec
+      @JsonProperty("parseSpec") ParseSpec parseSpec,
+      @JsonProperty("encoding") String encoding
   )
   {
     this.parseSpec = parseSpec;
     this.mapParser = new MapInputRowParser(parseSpec);
     this.parser = parseSpec.makeParser();
+
+    if(encoding != null) {
+      this.charset = Charset.forName(encoding);
+    } else {
+      this.charset = defaultCharset;
+    }
+  }
+
+  public StringInputRowParser(ParseSpec parseSpec)
```

Comment:
I guess this constructor is left to maintain backward compatibility. Can you please mark it deprecated?